EventReader/src/Event/EventStream.cs

```diff
@@ -0,0 +1,57 @@
+﻿using EventReader.Read;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+
+namespace EventReader.Event
+{
+    public class EventStream
+    {
+        IDataSource Source;
+
+        public EventStream(IDataSource source)
+        {
+            Source = source;
+        }
+
+        public IEnumerable<IEvent> Get(string query)
+        {
+            foreach(EventEntry entry in Reader.Read(Source, query)) // TODO: work out how to segment collection
+            {
+                yield return DeserializeEntry(entry);
+            }
+        }
+
+        public IEnumerable<IEvent> Get(string query, Dictionary<string, string> parameters)
+        {
+            foreach(EventEntry entry in Reader.Read(Source, query, parameters))
+            {
+                yield return DeserializeEntry(entry);
+            }
+        }
+
+        IEvent DeserializeEntry(EventEntry entry)
```

Comment:
It's good you pulled this one out, pragmatically it's probably fine for now, but if we needed to support many more event types we'd need to look at ways of separating the type resolving dependency from the deserialization (otherwise we'd end up with 100's of lines in a big if)